src/efftester.ml

```diff
@@ -185,11 +208,47 @@ let term_to_ocaml ?(typeannot = true) term =
       else Printf.bprintf sb "%s" x
     in
     match t with
+    | Constructor (_, name, payload_lst) ->
+      (match payload_lst with
+      | [] -> Printf.bprintf sb "%s" name
+      | trms ->
+        Printf.bprintf
+          sb
+          "(%s (%a))"
+          name
+          (* TODO: when we add support for payload that is a list with more than
+            one element, we need to fix the printing for payload list
+           *)
+            (fun sb lst -> List.iter (fun trm -> Printf.bprintf sb "%a" exp trm) lst)
+          trms)
+    | PatternMatch (_, match_trm, branches, _) ->
+      let rec print_pattern sb patt =
```

Comment:
This could/should be a toplevel function, mutually-recursive with `term_to_ocaml`, so it could be named `pattern_to_ocaml` as @jmid suggested previously. The reason to have it at the toplevel is that we may want to reuse it in other places than `PatternMatch` in the future (in OCaml, patterns are accepted in `fun p -> ` and `let p =`), and also that it makes the code mirror the structure of the grammar, which is nice.